go/mysqlconn/conn.go

```diff
@@ -0,0 +1,388 @@
+package mysqlconn
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"net"
+
+	"github.com/youtube/vitess/go/sqldb"
+
+	querypb "github.com/youtube/vitess/go/vt/proto/query"
+)
+
+const (
+	// connBufferSize is how much we buffer for reading and
+	// writing.
+	connBufferSize = 16 * 1024
+)
+
+// Conn is a connection between a client and a server, using the MySQL
+// binary protocol. It is built on top of an existing net.Conn, that
+// has already been established.
+//
+// Use Connect on the client side to create a connection.
+// Use NewListener to create a server side and listen for connections.
+type Conn struct {
```

Comment:
Conn will be the only one left eventually, I think. So I don't want a wrapper, I'll just delete stuff from sqldb_conn until it's gone, eventually.